app/javascript/leaflet/viewers/index_map.js

```diff
@@ -0,0 +1,176 @@
+            data.downloadUrl
+          }">
+            ${data.label ? data.label : "Download"}
+          </a>
+        </div>
+      </li>
+    </ul>
+  </li>
+`
+      : "";
+
+  indexMapTemplate = (data) => {
+    return new Promise((resolve) => {
+      if (data.iiifUrl && !data.thumbnailUrl) {
+        fetch(data.iiifUrl)
+          .then((response) => response.json())
+          .then((manifestResponse) => {
+            if (manifestResponse.thumbnail["@id"] !== null) {
+              data.thumbnailUrl = manifestResponse.thumbnail["@id"];
+            }
+            resolve(data); // Resolve the modified data
+          })
+          .catch(() => resolve(data)); // In case of error, resolve with original data
+      } else {
+        resolve(data); // Immediately resolve if no iiifUrl
+      }
+    });
+  };
+
+  availabilityStyle(availability) {
+    const options = JSON.parse(this.data.leafletOptions);
+    return availability || typeof availability === "undefined"
+      ? options.LAYERS.INDEX.DEFAULT
+      : options.LAYERS.INDEX.UNAVAILABLE;
+  }
+
+  addPreviewLayer() {
+    const options = JSON.parse(this.data.leafletOptions);
+    let prevLayer = null;
+
+    fetch(this.data.url)
+      .then((response) => response.json())
+      .then((data) => {
+        const geoJSONLayer = geoJson(data, {
+          style: (feature) =>
+            this.availabilityStyle(feature.properties.available),
+          onEachFeature: (feature, layer) => {
+            if (feature.properties.label !== null) {
+              layer.bindTooltip(feature.properties.label);
+            }
+            if (feature.properties.available !== null) {
+              layer.on("click", (e) => {
+                layer.setStyle(options.LAYERS.INDEX.SELECTED);
+                if (prevLayer !== null) {
+                  geoJSONLayer.resetStyle(prevLayer);
+                }
+                prevLayer = layer;
+                this.updateInformation(feature.properties);
+              });
+            }
+          },
+          pointToLayer: (feature, latlng) => circleMarker(latlng),
+        }).addTo(this.map);
+        this.map.fitBounds(geoJSONLayer.getBounds());
+      })
+      .catch((error) => console.error(error));
+  }
+
+  updateInformation = (properties) => {
+    this.indexMapTemplate(properties).then((updatedProperties) => {
+      const indexMapHtml = this.indexMapInfoTemplate(updatedProperties);
+      document.querySelector(".viewer-information").innerHTML = indexMapHtml;
+
+      const indexMapDownloadHtml =
+        this.indexMapDownloadTemplate(updatedProperties);
+      const downloadElement = document.querySelector(".js-download-list");
+
+      if (Boolean(downloadElement)) {
+        downloadElement.innerHTML += indexMapDownloadHtml; // Appends new download template
+        downloadElement
+          .querySelectorAll(".js-index-map-feature")
+          .forEach((elem) => elem.remove());
```

Comment:
old code did this before appending indexMapDownloadHtml to js-index-map-feature.